include/buffer_manager.h

```diff
@@ -222,7 +228,10 @@ namespace detail {
 		template <typename DataT, int Dims>
 		access_info access_host_buffer(buffer_id bid, access_mode mode, const subrange<Dims>& sr) {
 #if defined(CELERITY_DETAIL_ENABLE_DEBUG)
-			assert((m_buffer_types.at(bid)->has_type<DataT, Dims>()));
+			{
```

Comment:
Turns out `access_device_buffer` didn't lock the mutex for the same check since #162, and as there is no `access_device_buffer_impl` I decided to keep them there for symmetry (it's only for debug builds anyway).